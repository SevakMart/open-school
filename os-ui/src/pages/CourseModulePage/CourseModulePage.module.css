.ModuleOverviuw_container {
  display: grid;
  grid-template-columns: 30% 70%;
}/*# sourceMappingURL=CourseModulePage.module.css.map */