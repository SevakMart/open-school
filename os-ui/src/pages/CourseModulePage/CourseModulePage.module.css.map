{"version":3,"sources":["CourseModulePage.module.scss","CourseModulePage.module.css"],"names":[],"mappings":"AAAA;EACI,aAAA;EACA,8BAAA;ACCJ","file":"CourseModulePage.module.css"}