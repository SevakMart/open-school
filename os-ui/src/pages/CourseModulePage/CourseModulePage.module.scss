@import '../../styles/Configs/Fonts.scss';
@import url($montserrat);

.ModuleOverviuw_container {
    display: grid;
    grid-template-columns: 30% 70%;
    font-family: 'Montserrat', sans-serif;
}