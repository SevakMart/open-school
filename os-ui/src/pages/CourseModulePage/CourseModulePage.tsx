/* eslint-disable implicit-arrow-linebreak */
/* eslint-disable array-callback-return */
import { useMemo, useState } from 'react';
import { userContext } from '../../contexts/Contexts';
import CourseModuleSidebar from './Subcomponent/CourseModuleSidebar/CourseModuleSidebar';
import styles from './CourseModulePage.module.scss';
import ModuleMainPage from './Subcomponent/ModuleM1Page/ModulMainPage';
import NavbarOnSignIn from '../../component/Navbar-Component/NavbarOnSignIn/NavbarOnSignIn';

const CourseModulePage = ({ userInfo }:{userInfo:any}) => {
  const idAndToken = useMemo(() => ({
	  token: (userInfo as any).token,
	  id: (userInfo as any).id,
  }), []);

  	const [value, setValue] = useState<string>('1');

  const handleChangeValue = () => {
    setValue(value);
  };

  return (
    <>
      <NavbarOnSignIn />
      <userContext.Provider value={idAndToken}>
        <div className={styles.ModuleOverviuw_container}>
          <CourseModuleSidebar value={value} handleChangeValue={handleChangeValue} />
          <ModuleMainPage value={value} handleChangeValue={handleChangeValue} />
        </div>
      </userContext.Provider>
    </>
  );
};
export default CourseModulePage;
