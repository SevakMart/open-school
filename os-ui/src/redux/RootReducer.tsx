import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
