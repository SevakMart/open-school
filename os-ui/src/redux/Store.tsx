import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
