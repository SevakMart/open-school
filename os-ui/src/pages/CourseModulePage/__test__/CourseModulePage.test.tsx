import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../redux/Store';
import CourseModulePage from '../CourseModulePage';

/* eslint-disable max-len */

const userInfo = {
  token: 'fake token',
  id: 1,
};
jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key:string) => key }),
}));

describe('Create test cases for course overview page', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}><CourseModulePage userInfo={userInfo} /></Provider>,
	 { wrapper: BrowserRouter },
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
