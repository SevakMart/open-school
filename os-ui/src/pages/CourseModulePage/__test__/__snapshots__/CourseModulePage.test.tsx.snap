// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for course overview page Create a snapshot test 1`] = `<DocumentFragment />`;
